### DISCUSSION TITLE
Question on `serverTimestamp` being a function

### DISCUSSION THREAD

**Discussion by akauppi** [firebase/firebase-js-sdk#4569] [Q&A]
Disclaimer: I'm *not* convinced making `serverTimestamp` not be a function is a good thing. What do you others think?

---

In the Firebase Firestore 8.x API we do:

```
FieldValue.serverTimestamp()
```

I've never stopped to think why that's a function, but now I did, with the `@exp` API. I somehow expected the parantheses to have been dropped, but they weren't. It's still:

```
import { serverTimestamp } from 'firebase/firestore'

...

serverTimestamp()
```
 
Of the five `FieldValue`s, `arrayRemove`, `arrayUnion` and `increment` need parameters. They are generators for the actual sentinels. `serverTimestamp` and `delete` don't have parameters, and I presume there's always the same sentinel returned by them.

This is not a big thing. But.. interfaces also convey the meaning to us humans. To me, exporting the sentinels themselves would make sense, from an interface clarity point of view.

Is there a reason why `serverTimestamp` needs to be called?

**Comment by akauppi** [comment]
Related to this, and naming is the `delete` sentinel function. It used to be used as `FieldValue.delete` but now dropping the `FieldValue` makes it a bit too generic, imho.

I'd like to give a better suggestion but cannot.

**Reply by schmidt-sebastian** [reply]
`delete()` should already be called `deleteField()`. Is that still too generic?

**Reply by akauppi** [reply]
Of course that name is fine. I had not noticed - thanks for correcting me.

How about the subject's question on `()` or not?

**Reply by schmidt-sebastian** [reply]
We are discussing that internally. Thanks for bringing it up.
